.github/workflows/cherry-pick.yml

```diff
@@ -31,320 +31,321 @@ env:
-        run: git rm ${{ steps.changelog.outputs.changelogsToBeDeleted }}
-
-      - name: Commit changes for deletion
-        if: steps.changelog.outputs.changelogsToBeDeleted != '' && steps.changelog.outputs.changelogsToBeDeleted != null
-        run: git commit --no-verify -am "Delete changelog files for ${{ needs.prep.outputs.pr }}"
-
-      - name: Push deletion branch up
-        if: steps.changelog.outputs.changelogsToBeDeleted != '' && steps.changelog.outputs.changelogsToBeDeleted != null
-        run: git push origin delete-changelogs/${{ needs.prep.outputs.pr }}
-
-      - name: Create the PR for deletion branch
-        id: deletion-pr
-        if: steps.changelog.outputs.changelogsToBeDeleted != '' && steps.changelog.outputs.changelogsToBeDeleted != null
-        uses: actions/github-script@v6
-        with:
-          script: |
-            const pr = await github.rest.pulls.create({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              title: "Delete changelog files based on PR ${{ needs.prep.outputs.pr }}",
-              head: "delete-changelogs/${{ needs.prep.outputs.pr }}",
-              base: "trunk",
-              body: "Delete changelog files based on PR #${{ needs.prep.outputs.pr }}"
-            })
-
-            console.log( `::set-output name=deletion-pr::${ pr.data.html_url }` )
-
-      - name: Notify Slack on failure
-        if: ${{ failure() && inputs.skipSlackPing != true }}
-        uses: archive/github-actions-slack@v2.0.0
-        with:
-          slack-bot-user-oauth-access-token: ${{ secrets.CODE_FREEZE_BOT_TOKEN }}
-          slack-channel: ${{ inputs.slackChannelOverride || secrets.WOO_RELEASE_SLACK_CHANNEL }}
-          slack-text: |
-            :warning-8c: Code freeze violation. PR(s) created that breaks the Code Freeze for '${{ needs.prep.outputs.release }}' :ice_cube:
-
-            An attempt to cherry pick PR(s) into outgoing release '${{ needs.prep.outputs.release }}' has failed. This could be due to a merge conflict or something else that requires manual attention. Please check: https://github.com/woocommerce/woocommerce/pull/${{ needs.prep.outputs.pr }}
-
-      - name: Notify Slack on success
-        if: ${{ success() && inputs.skipSlackPing != true }}
-        uses: archive/github-actions-slack@v2.0.0
-        with:
-          slack-bot-user-oauth-access-token: ${{ secrets.CODE_FREEZE_BOT_TOKEN }}
-          slack-channel: ${{ inputs.slackChannelOverride || secrets.WOO_RELEASE_SLACK_CHANNEL }}
-          slack-text: |
-            :warning-8c: Code freeze violation. PR(s) created that breaks the Code Freeze for '${{ needs.prep.outputs.release }}' :ice_cube:
-            
-            Release lead please review:
-
-            ${{ steps.cherry-pick-pr.outputs.cherry-pick-pr }}
-            ${{ steps.deletion-pr.outputs.deletion-pr }}
+
+                      core.setOutput( 'changelogsToBeDeleted', changelogsToBeDeleted )
```

Comment:
```suggestion
                      core.setOutput( 'changelogsToBeDeleted', changelogsToBeDeleted.join( ' ' ) )
```

Reply:
With the change to use `setOutput`, I think we need to convert the array to string to be able to consume it later. Otherwise when we reach the logic for `git rm`, it will try `git rm []` when no changelog files and that will cause an error.

Reply:
@roykho in JavaScript an empty array coerced to a string is just `''` so there is no need to do that if your goal is to ensure an empty string?

Reply:
@samueljseay - you can test it yourself. It will not work without that change. Test with changelog, without changelog and with multiple changelogs.

Reply:
@roykho ok I wasn't clear that you'd actually tested that.

I see now the issue. `core.setOutput` does a JSON.stringify of the value which is why we get `[]` Yep your solution looks good, and will handle both the empty and non-empty state. Thanks!

Reply:
That's resolved now.